www/matj27/sp2/js/app.js

```diff
@@ -0,0 +1,263 @@
+$(document).ready(() => {
+    App.init();
+});
+
+window.onpopstate = function (event) {
+    App.lastLocationPoppedFromHistory = true;
+
+    const lon = App.urlParameter('lon');
+    const lat = App.urlParameter('lat');
+    App.weatherUnits = App.urlParameter('units');
+
+    const newCoords = SMap.Coords.fromWGS84(lon, lat)
+    Mapycz.createMarker(newCoords);
+    Mapycz.map.setCenter(newCoords);
+
+    OpenWeather.retrieveWeather(lat, lon);
+};
+
+const App = {
+    openWeatherLocalStorageKey: 'open-weather-api-key',
+    sendGridLocalStorageKey: 'send-grid-api-key',
+    appDiv: $('#app'),
+    emailDivRendered: false,
+    weatherDivRendered: false,
+    weatherUnits: 'metric',
+    weatherLang: 'cz',
+    lastLocationPoppedFromHistory: false
+};
+
+/**
+ * Runs initialization procedures for map view and for email sender form
+ */
+App.init = async () => {
+    App.emailDivRendered = false;
+    App.weatherDivRendered = false;
+
+    let appStructurePrepared = false;
+    while (!appStructurePrepared) {

```

Comment:
neni treba pretezovat browser aby timto zpusobem loopoval cekal na data, takhle nikdo nepouziva async await, ty slouzi predevsim pro Promisove objekty, lepsi by bylo to udelat interaktivne,